packages/block-editor/src/components/block-list/block-selection-button.js

```diff
@@ -262,7 +288,7 @@ function BlockSelectionButton( { clientId, rootClientId, blockElement } ) {
 				ref={ ref }
 				onClick={ () => setNavigationMode( false ) }
 				onKeyDown={ onKeyDown }
-				label={ label }
+				describedBy={ label }
 			>
 				<BlockTitle clientId={ clientId } />
```

Comment:
Sorry, one more thing. I think this should always return the proper title of the block, e.g. 'Paragraph'. But some blocks are able to override this using the `__experimentalLabel` function and return a different value. For sighted users this is useful in a few features (List View, breadcrumbs) to discern blocks with the same title, especially where there might be a lot of the same block used as siblings.

The best example of this is the Link block that's used in the Navigation block ([code here](https://github.com/WordPress/gutenberg/blob/aec1ddb17009dcbc82e8b02ee4bb889a148338a2/packages/block-library/src/navigation-link/index.js#L29)). BlockTitle will always return its content (from its label attribute), so if I type 'Home' for a link, BlockTitle returns 'Home'. `getAccessibleLabel` would also output 'Home' through the label value. The actual block title isn't read at all.

For navigation mode it's probably better always have the proper block title here. I don't think there's any advantage to using some of the block content, as this button is directly above the content visually.

Reply:
I believe I removed title changes. Should be okay.